examples/echo_server.hs

```diff
@@ -0,0 +1,18 @@
+{-# OPTIONS -XOverloadedStrings #-}
+
+import qualified Network.Freddy as Freddy
+import Data.ByteString.Lazy.Char8 (ByteString)
+
+processMessage :: Freddy.Request -> IO ()
+processMessage (Freddy.Request body replyWith failWith) = replyWith body
+
+main = do
+  fConnection <- Freddy.connect "127.0.0.1" "/" "guest" "guest"
```

Comment:
So the question is if the `respondTo` should take `IO Connection` or `Connection` as an argument. I don't have a good explanation. I usually avoid passing monads if I don't have to. AMQP library itself uses the same style https://github.com/hreinhardt/amqp/blob/master/Network/AMQP.hs#L17-L18 .

Maybe @urmastalimaa can give an explanation.